pkg/extendapi/devicetemplate.go

```diff
@@ -0,0 +1,216 @@
+package extendapi
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/cnrancher/edge-api-server/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/tidwall/gjson"
+
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+
+	// appsv1 "k8s.io/api/apps/v1"
+	yaml2 "github.com/ghodss/yaml"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type templateData struct {
+	Template struct {
+		Metadata struct {
+			Labels map[string]string `yaml:"labels"`
+		} `yaml:"metadata"`
+	} `yaml:"template"`
+}
```

Comment:
you may define a `deviceTemplate` struct with specified device apiVersion, kind, template metadata and rawExtenstion spec.